src/processor/operator/persistent/reader.cpp

```diff
@@ -84,7 +80,7 @@ void Reader::readNextDataChunk() {
         readFunc(*readFuncData, morsel->blockIdx, dataChunk.get());
         if (dataChunk->state->selVector->selectedSize > 0) {
             leftArrowArrays.appendFromDataChunk(dataChunk.get());
-        } else {
+        } else if (readFuncData->emptyBlockImpliesDone()) {
             sharedState->moveToNextFile();
```

Comment:
We definitely need this. Otherwise, lines larger than the PARALLEL_BLOCK_SIZE will die. Or, if the header is long enough, we will parse incorrectly.

The specific reason is that if a line is very long and the block end is within this line, the parallel CSV reader will have no data for this block. It will thus return zero. However, that does **not** mean we should move to the next file, in fact, it would be an error to do so.

I can add a test for this if you like.

Reply:
Can you add a comment about the reason?

Reply:
Yes, and Ill add a test as well later today.